src/js/components/Button/__tests__/Button-kind-test.js

```diff
@@ -0,0 +1,277 @@
+
+  test('custom theme secondary', () => {
+    const { container } = render(
+      <Grommet theme={buttonKindTheme}>
+        <Button secondary />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('border', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              border: {
+                color: 'green',
+                width: '2px',
+              },
+            },
+          },
+        }}
+      >
+        <Button label="Test" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('no border', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              border: false,
+            },
+          },
+        }}
+      >
+        <Button label="Test" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('extend', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              extend: {
+                color: 'green',
+              },
+            },
+          },
+        }}
+      >
+        <Button label="Test" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('fill', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: { default: {} },
+        }}
+      >
+        <Button label="Test" fill />
+        <Button label="Test" fill="vertical" />
+        <Button label="Test" fill="horizontal" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('font', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              font: {
+                weight: 700,
+                height: '20px',
+              },
+            },
+          },
+        }}
+      >
+        <Button label="Test" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('font undefined', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              font: {
+                weight: undefined,
+                height: undefined,
+              },
+            },
+          },
+        }}
+      >
+        <Button label="Test" />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('hover', () => {
+    const { container } = render(
+      <Grommet
+        theme={{
+          button: {
+            default: {
+              hover: {
+                color: 'white',
+                background: 'green',
+              },
+            },
+          },
+        }}
+      >
+        <Button label="Test" plain hoverIndicator onClick={() => {}} />
```

Comment:
Might it be better to test this with a `mouseOver` event instead of just setting `hoverIndicator` to true so that we can test the behavior as well?

Reply:
I don't think the mouseOver would change anything in the snapshot because the CSS for the hover in the snapshot is:
```
.c1:hover {
  background-color: rgba(221,221,221,0.4);
  color: #444444;
}
```
The CSS and the shapshot wouldn't change if there was a mouseOver or not. Although the user would see a change, the code in the snapshot wouldn't be any different.